projects/common/src/telemetry/track/track.directive.ts

```diff
@@ -0,0 +1,17 @@
+import { Directive, HostListener, Input } from '@angular/core';
+import { UserTelemetryInternalService } from '../user-telemetry-internal.service';
+
+@Directive({
+  selector: '[htTrack]'
+})
+export class TrackDirective {
+  @Input('htTrack')
+  public name!: string;
+
+  public constructor(private readonly userTelemetryInternalService: UserTelemetryInternalService) {}
+
+  @HostListener('click', ['$event'])
+  public trackClick(event: MouseEvent): void {
+    this.userTelemetryInternalService.trackEvent(`Click: ${this.name}`, { target: event.target, type: event.type });
```

Comment:
what's the name meant to be here?

Reply:
yeah, so the name could be anything we configure.  The product team can specify what name they want for a particular click event

We can also set it to "Logout" when we use this directive on the logout button. We can also add this directive on the button component and use the button label as `name`.

Reply:
I'd probably take it as a directive arg

Reply:
It is a directive argument
  `@Input('htTrack') public name!: string;`

Reply:
ah whoops - went straight to the listener and missed that.